Assets/Mirage/Runtime/NetworkSceneManager.cs

```diff
@@ -19,161 +19,152 @@ namespace Mirage
+            Client.MessageHandler.RegisterHandler<NotReadyMessage>(ClientNotReadyMessage);
         }
 
-        #region Client
-
-        void RegisterClientMessages()
-        {
-            Client.MessageHandler.RegisterHandler<SceneMessage>(ClientSceneMessage);
-            if (!Client.IsLocalClient)
-            {
-                Client.MessageHandler.RegisterHandler<SceneReadyMessage>(ClientSceneReadyMessage);
-                Client.MessageHandler.RegisterHandler<NotReadyMessage>(ClientNotReadyMessage);
-            }
-        }
-
-        internal void ClientSceneMessage(INetworkPlayer player, SceneMessage msg)
+        /// <summary>
+        ///     Received message from server to start loading scene or scenes.
+        ///
+        ///     <para>Default implementation is to load main activate scene server is using and load any
+        ///     other additive scenes in background and notify event handler. If this is not intended
+        ///     behavior you need please override this function.</para>
+        /// </summary>
+        /// <param name="player"></param>
+        /// <param name="message"></param>
+        public virtual void ClientStartSceneMessage(INetworkPlayer player, SceneMessage message)
         {
             if (!Client.IsConnected)
-            {
-                throw new InvalidOperationException("ClientSceneMessage: cannot change network scene while client is disconnected");
-            }
-            if (string.IsNullOrEmpty(msg.scenePath))
-            {
-                throw new ArgumentNullException(nameof(msg.scenePath), $"ClientSceneMessage: {nameof(msg.scenePath)} cannot be empty or null");
-            }
+                throw new InvalidOperationException("[NetworkSceneManager] - SceneLoadStartedMessage: cannot change network scene while client is disconnected");
 
-            if (logger.LogEnabled()) logger.Log($"ClientSceneMessage: changing scenes from: {ActiveScenePath} to: {msg.scenePath}");
+            if (string.IsNullOrEmpty(message.MainActivateScene))
+                throw new ArgumentNullException(nameof(message.MainActivateScene), $"[NetworkSceneManager] - SceneLoadStartedMessage: {nameof(message.MainActivateScene)} cannot be empty or null");
+
+            if (logger.LogEnabled()) logger.Log($"[NetworkSceneManager] - SceneLoadStartedMessage: changing scenes from: {ActiveScenePath} to: {message.MainActivateScene}");
 
             //Additive are scenes loaded on server and this client is not a host client
-            if (msg.additiveScenes != null && msg.additiveScenes.Length > 0 && Client && !Client.IsLocalClient)
+            if (message.AdditiveScenes != null && message.AdditiveScenes.Length > 0 && Client && !Client.IsLocalClient)
             {
-                foreach (string scene in msg.additiveScenes)
+                for (int index = message.AdditiveScenes.Length - 1; index >= 0; index--)
```

Comment:
why are we loading these in reverse order?

Reply:
Habit of doing things in reverse order for collections. Does it really matter the order tho?